src/arrays/arraymath.jl

```diff
@@ -121,6 +121,11 @@ function \{B<:OrthonormalBasis}(op::AbstractQuMatrix{B}, vec::AbstractQuVector{B
     return QAT(div, bases(op,1))
 end
 
+# dot product of QuArray's
+dot{B<:OrthonormalBasis}(vec1::AbstractQuVector{B}, vec2::AbstractQuVector{B}) = dot(coeffs(vec1), coeffs(vec2))
```

Comment:
Do we need extra definitions for the cases where one vector is actually a dual vector in which case we should throw an error?